include/hipSYCL/sycl/libkernel/cuda/group_functions.hpp

```diff
@@ -52,12 +52,17 @@ T group_broadcast(sub_group g, T x,
 // barrier
 template<typename Group>
 HIPSYCL_KERNEL_TARGET
-inline void group_barrier(Group g, memory_scope fence_scope = Group::fence_scope) {
+inline void group_barrier(Group g, memory_scope fence_scope) {
```

Comment:
What do we need this template argument for? Couldn't we just overload with `sycl::group<int Dim>` instead of allowing pretty much anything as argument? After all, this barrier implementation assumes work groups, right?

Reply:
Originally it was to stay in line with the signature presented in the specification. We could adjust it, but then I would do it for all group functions, which have specializations for sub_groups to keep it consistent.

Reply:
I think this might be a step that we might have to take in light of the fact that we will soon get yet another work group type for scoped parallelism ;)

Do you want to do this change in this PR or in a subsequent one? I'm fine with either.

Reply:
I think a second PR would make it easier to check the diff for errors.

Reply:
Should I use `int Dim` in the template in combinaion with `sycl::group<DIM>` or, as it is currently done for the joint_* functions, `std::enable_if_t<is_group_v<std::decay_t<Group>>, bool> = true`?
Also the rename PR makes sense to be merged after #607, so I can rename on the already renamed functions.

Reply:
Overload for `sycl::group` specifically, please. The second option is going to break when scoped parallelism 2.0 hits, which already comes with its own `group_barrier` overloads for the new `sp_group` type :)

> Also the PR adjusting the templates makes sense to be merged after #607, so I can make the changes on the already renamed functions.

Okay, so we'll pull in #607 and then #617.